diff --git a/Makefile b/Makefile
index d6ea366f..8aa27e10 100644
--- a/Makefile
+++ b/Makefile
@@ -139,7 +139,7 @@ ABI_FLAGS := $(ARCH_ABI_FLAGS) -ffreestanding -fno-builtin \
 	-fno-stack-protector -fomit-frame-pointer
 LINK_FLAGS = $(ARCH_LINK_FLAGS) $(ABI_FLAGS) -fuse-ld=bfd \
 	-Wl,-T,$(LDSCRIPT_OBJ) -Wl,--gc-sections -Wl,-Map=$@.map
-CFLAGS := $(ARCH_CFLAGS) -Wall -Werror -Wstrict-prototypes -Wshadow \
+CFLAGS := $(ARCH_CFLAGS) -Wall -Wno-error -Wstrict-prototypes -Wshadow \
 	-Wno-address-of-packed-member \
 	$(INCLUDES) -std=gnu99 $(ABI_FLAGS) -ffunction-sections -fdata-sections -ggdb3
 
