diff --git a/src/debug/firmware_shell/storage.c b/src/debug/firmware_shell/storage.c
index 97b67cda..7e00aee5 100644
--- a/src/debug/firmware_shell/storage.c
+++ b/src/debug/firmware_shell/storage.c
@@ -45,8 +45,6 @@ static int storage_read(int argc, char *const argv[])
 	int *args[] = {&base_block, &num_blocks, (int*) &dest_addr};
 	BlockDev *bd;
 
-	dest_addr = NULL;
-
 	for (i = 0; i < ARRAY_SIZE(args); i++)
 		*args[i] = strtoul(argv[i], NULL, 0);
 
